### ISSUE TITLE
opaque types are not FFI-safe

### ISSUE THREAD

**Issue by marrub--** [maia-s/sdl3-sys-rs#8]
```rs
unsafe extern "C" {
    fn thingy(window: *mut sdl3_sys::video::SDL_Window);
}
```
```
warning: `extern` block uses type `SDL_Window`, which is not FFI-safe
   = note: this struct is non-exhaustive
   = note: `#[warn(improper_ctypes)]` on by default
```
the warning in this case is only menacing and not really meaningful but i would rather not have to disable it

since the body is `_opaque: [::core::primitive::u8; 0]`, do these even need `#[non_exhaustive]`?

**Comment by Lokathor** [comment]
The lint is just being overly conservative, to the point of being wrong.

However, separately, I'd also suggest `c_void` instead of a 0 length array, just for clarity of intent.

**Comment by maia-s** [comment]
I can remove non_exhaustive from the opaque structs, and use c_void instead of the zero length array. I'm not sure why I added non_exhaustive to those, to be honest. I'll commit a fix for that later and release 0.1.2 with the fix.

However, I do use non_exhaustive for some other structs deliberately. For example, `SDL_Surface` is actually bigger internally in SDL than the public definition, so if you construct one yourself and pass it to SDL it's instant UB. I added non_exhaustive to guard against that.

I'm surprised about this warning though. I don't get any warnings about this when building or running clippy on sdl3-sys itself, and I don't disable that warning. Also, non_exhaustive doesn't change the layout of a type, so there's no reason it shouldn't be ffi safe. For structs it just prevents them from getting constructed with struct init syntax. Maybe we should post an issue about this at the rust github?
